gap/congruences/cong.gi

```diff
@@ -21,9 +21,32 @@
 ## cong.gd contains declarations for many of these.
 ##
 
+InstallMethod(\=, "for two semigroup congruences",
+[IsSemigroupCongruence, IsSemigroupCongruence],
+function(c1, c2)
+  if not IsFinite(Range(c1)) then
+    TryNextMethod();
+  fi;
+  return Range(c1) = Range(c2)
+         and EquivalenceRelationCanonicalLookup(c1) =
+             EquivalenceRelationCanonicalLookup(c2);
+end);
+
+InstallMethod(\=, "for two semigroup congruences with generating pairs",
+[IsSemigroupCongruence and HasGeneratingPairsOfMagmaCongruence,
+ IsSemigroupCongruence and HasGeneratingPairsOfMagmaCongruence],
+function(c1, c2)
+  return Range(c1) = Range(c2)
+         and ForAll(GeneratingPairsOfSemigroupCongruence(c1), p -> p in c2)
+         and ForAll(GeneratingPairsOfSemigroupCongruence(c2), p -> p in c1);
+end);
+
 InstallMethod(\=, "for a left and a right semigroup congruence",
 [IsLeftSemigroupCongruence, IsRightSemigroupCongruence],
 function(c1, c2)
```

Comment:
It could be that they are both two-sided congruences, but specified using different sets of generating pairs (one using `LeftSemigroupCongruence` and the other using `RightSemigroupCongruence`).